libraries/bezier-rs/src/subpath/lookup.rs

```diff
@@ -10,6 +12,70 @@ impl Subpath {
 		self.iter().fold(0., |accumulator, bezier| accumulator + bezier.length(num_subdivisions))
 	}
 
+	fn global_euclidean_to_local_euclidean(&self, global_t: f64) -> (usize, f64) {
+		let lengths = self.iter().map(|bezier| bezier.length(None)).collect::<Vec<f64>>();
+		let total_length: f64 = lengths.iter().sum();
+
+		let mut accumulator = 0.;
+		for (index, length) in lengths.iter().enumerate() {
+			let length_ratio = length / total_length;
+			if accumulator <= global_t && global_t <= accumulator + length_ratio {
+				return (index, (global_t - accumulator) / length_ratio);
+			}
+			accumulator += length_ratio;
+		}
+		(0, 0.)
+	}
+
+	/// Convert a [SubpathTValue] to a parametric `(segment_index, t)` tuple.
+	/// - Asserts that `t` values contained within the `SubpathTValue` argument lie between 0 and 1.
+	/// - If the argument is a variant containing a `segment_index`, asserts that the index references a valid segment on the curve.
+	pub(crate) fn t_value_to_parametric(&self, t: SubpathTValue) -> (usize, f64) {
+		assert!(self.len() > 1);
```

Comment:
Why would we want to panic if the subpath is composed of just one segment?

Reply:
`self.len()` is actually just a shortcut for `self.manipulator_groups.len`, so this is actually checking to make sure the number of segments is non-zero. `self.len_segments()` counts the number segments. I use that function instead to make this clearer.